MetaMask/metamask-extension
# Replaced addresses by the address component on SignTypedData v4 signatures

## Explanation

MetaMask has no control on what the message that is displayed is this screen is - this is something provided by the dapps and will look differently for each one of them. MetaMask does have control on how such messages are displayed and how the UI looks like - and this is what we are trying to achieve with this PR. So with this PR, we want every value that is an address to be displayed using the address component. 

## More Information

* Fixes #15716 

## Screenshots

### Before (test-dapp)

![Screenshot 2022-10-04 at 14 32 15](https://user-images.githubusercontent.com/92527393/193820100-0d10f36c-0344-473d-9ee8-644c0676d716.png)

![Screenshot 2022-10-04 at 14 32 29](https://user-images.githubusercontent.com/92527393/193820114-15ace0be-8214-400c-90ed-ea8a56ba0c8e.png)

### After (test-dapp)

![Screenshot 2022-10-12 at 10 08 11](https://user-images.githubusercontent.com/92527393/195301469-ba7dd559-cfb1-4840-97e7-4a58c22da64e.png)

![Screenshot 2022-10-12 at 10 08 51](https://user-images.githubusercontent.com/92527393/195301498-3e739079-8e42-4e36-8eca-8cf870c6b073.png)

## Screencaps

### Before (test-dapp)

https://user-images.githubusercontent.com/92527393/192788293-5411b612-b8fb-4009-83d9-5c4a856712a7.mov

### After (test-dapp)

https://user-images.githubusercontent.com/92527393/193854495-4df94bf0-8f41-4d1c-9336-e00b620d3248.mov

### Before (OpenSea)

https://user-images.githubusercontent.com/92527393/193039323-341317c1-509b-4a63-a5a7-29b36aa2d36c.mov

### After (OpenSea)

https://user-images.githubusercontent.com/92527393/193858085-7b00dc45-c459-4fcf-8692-3d3e952be93f.mov

## Manual Testing Steps

1. Open the test dapp https://metamask.github.io/test-dapp/
2. Under the SignTypedData v4 section, click on `Sign`
    or
3. Get some test `NFTs` on https://faucet.paradigm.xyz/
4. Make sure MM is connected to `Rinkeby test network`
5. Go to https://testnets.opensea.io/account
6. Click on one of your `NFTs`
7. Click on `Sell`
8. Go through the steps until you get prompt to `Confirm listing` on OpenSea - this will trigger a `SignTypedData v4` prompt in MM (you might have to sign a different message and approve a transaction if it's the first time you're going through this flow)
